protocols/imap/src/test/java/org/apache/james/imap/encode/ListingEncodingUtilsTest.java

```diff
@@ -42,7 +43,7 @@
 
     @Test
     public void encodeShouldWriteNilDelimiterWhenUnassigned() throws Exception {
-        ListResponse input = new ListResponse(Children.HAS_CHILDREN, Selectability.NONE, nameParameter, new PathDelimiterImpl(((char) Character.UNASSIGNED)));
+        ListResponse input = new ListResponse(Children.HAS_CHILDREN, Selectability.NONE, nameParameter, NoPathDelimiter.SINGLETON);
```

Comment:
the problem with two two classes is that NoPathDelimiter is nether isntanciated. If I create a PathDelimiter with Character.UNASSIGNED, it just throws instead of creating NoPathDelimiter.
Furthermore, it demonstrates that the refactoring is done blindly : we don't have a high-level test case that covers this change (and we should).

Reply:
So... 

 - Static method in the interface that gives you the Path
 - I disagree with the throw on join but anyway. If you wish. After all, that's never used so I don't care.
 - So you seriously think James ever worked with no delimiter? What do you mean by "we don't have a high-level test case that covers this change (and we should)"

Do you mean that I should write MPT tests when delimiter is "\0"? And ensure James works well when that happen? If so, I don't see why I would continue losing time on that issue. 

Moreover, that would mix **refactoring** and **feature addition** IMO. Does handling "\0" stuff add value in the first place?

Isn't it the definition of refactoring: **not changing high level, protocol layer tests**?